tests/framework/test_engine.py

```diff
@@ -347,6 +348,17 @@ def test_SimulationContext_report_write(SimulationContext, base_config, componen
         assert results.equals(written_results)
 
 
+@pytest.mark.skip(reason="TODO: Figure out how to make Dill serialize in pytest")
+def test_SimulationContext_write_backup(SimulationContext, tmpdir):
+    sim = SimulationContext()
+    backup_path = tmpdir / "backup.pkl"
+    sim.write_backup(backup_path)
+    assert backup_path.exists()
+    with open(backup_path, "rb") as f:
+        sim_backup = dill.load(f)
+    assert isinstance(sim_backup, SimulationContext)
+
```

Comment:
Would there be a way you could test that a backup is saved up until failure? This might be more a VCT but that is a test we will definitely want.

Reply:
Since I refactored sim.run(), I added a test that we save a certain number of times while the simulation is running. Does that capture the behavior you're looking for or do you think we need to add an explicit execution error somewhere and check against that?